search-api/search-core/src/main/java/org/sunbird/search/processor/SearchProcessor.java

```diff
@@ -902,14 +902,26 @@ private QueryBuilder getQuery(SearchDTO searchDTO) {
 		return prepareSearchQuery(searchDTO);
 	}
 
-	private static QueryBuilder getPostFilterQuery(String orgId) {
+	private static QueryBuilder getPostFilterQuery(Map<String, Object> postFilter) {
 		// Creating the post_filter bool query
 		BoolQueryBuilder postFilterBoolQuery = QueryBuilders.boolQuery();
-		// Nested query for "secureSettings"
+		BoolQueryBuilder nestedBoolQuery = QueryBuilders.boolQuery();
+
+		for (Map.Entry<String, Object> filters : postFilter.entrySet()) {
+			if (filters.getValue() instanceof List) {
+				for (String value : (List<String>) filters.getValue()) {
+					nestedBoolQuery.should(QueryBuilders.termQuery(filters.getKey(), value));
+				}
+			} else if (filters.getValue() instanceof String) {
+				nestedBoolQuery.should(QueryBuilders.termQuery(filters.getKey(), ((String) filters.getValue()).toLowerCase()));
+			} else {
```

Comment:
Need to throw error -- when type is not list or string - instead of blind handling ?

Reply:
Done